websites/A/AniMixPlay/presence.ts

```diff
@@ -0,0 +1,111 @@
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused",
+    browsing: "presence.activity.browsing"
+});
+
+/**
+ * Get Timestamps (Credit to Bas950)
+ * @param {Number} videoTime Current video time seconds
+ * @param {Number} videoDuration Video duration seconds
+ */
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
+const browsingTimer = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+    const presenceData: presenceData = {
+        largeImageKey: "logo-v2"
+    };
+
+    if (document.location.hostname == "animixplay.to") {
+      if (document.location.pathname == "/") {
+
+        urlParams = new URLSearchParams(window.location.search);
+
+        if (urlParams.get("q")) {
+          presenceData.startTimestamp = browsingTimer;
+          presenceData.details = "Currently searching for...";
+          presenceData.state = '"' + urlParams.get("q") + '"';
+          presenceData.smallImageKey = "browsing-v1";
+          presenceData.smallImageText = (await strings).browsing;
+        } else if (urlParams.get("genre")) {
+            presenceData.startTimestamp = browsingTimer;
+            presenceData.details = "Currently exploring...";
+            presenceData.state = "" + urlParams.get("genre").toLowerCase().replaceAll(",", " + ") + " related anime";
+            presenceData.smallImageKey = "browsing-v1";
+            presenceData.smallImageText = "Exploring...";
+        } else if (urlParams.get("season") && urlParams.get("year")) {
+            presenceData.startTimestamp = browsingTimer;
+            presenceData.details = "Currently exploring...";
+            presenceData.state = "" + urlParams.get("season").toLowerCase() + " " + urlParams.get("year").toLowerCase() + " anime";
+            presenceData.smallImageKey = "browsing-v1";
+            presenceData.smallImageText = "Exploring...";
+        } else {
+            presenceData.startTimestamp = browsingTimer;
+            presenceData.details = "Currently browsing...";
+            presenceData.smallImageKey = "browsing-v1";
+            presenceData.smallImageText = (await strings).browsing;
+        }
+      } else if (document.location.pathname.includes("/v4/") || document.location.pathname.includes("/v1/") || document.location.pathname.includes("/v7/")) {
+        let currentTime: any,
+          duration: any,
+          paused: any,
+          timestamps: any,
+          video: HTMLVideoElement;
+        
+        const iframe = document.getElementById("iframeplayer");
+        if (iframe) {
+          video = iframe.contentWindow.document.getElementsByTagName("video")[0];
+        } else {
+          video = document.querySelector("#playercontainer > div > div.plyr__video-wrapper > video");
+        }
```

Comment:
if you get iframe, use iframe.ts thanks :)

Reply:
Migrated :)